### ISSUE TITLE
vcpkg: crash in vcpkg::System::cmd_execute_clean

### ISSUE THREAD

**Issue by Mixaill** [Microsoft/vcpkg#1648]
Windows 10 15063.540
MSVS 2017 15.3, v140 toolset, 8.1 target platform
Vcpkg git master 45fcdbc

Only in "Release" configuration, 32-bit and 64-bit

---

https://github.com/Microsoft/vcpkg/blob/master/toolsrc/src/vcpkg_System.cpp#L120
```
Unhandled exception at 0x75B7AC9B (ucrtbase.dll) in vcpkg.exe: 0xC0000005: Access violation reading location 0x01324000.
```

call stack
```
 	ucrtbase.dll!__crt_char_traits<wchar_t>::tcslen<wchar_t * &>(wchar_t * &)
 	ucrtbase.dll!<lambda>(void)()
 	ucrtbase.dll!construct_environment_block<wchar_t>()
 	ucrtbase.dll!common_pack_argv_and_envp<wchar_t>()
 	ucrtbase.dll!execute_command<wchar_t>()
 	ucrtbase.dll!common_spawnv<wchar_t>()
 	ucrtbase.dll!common_spawnvp<wchar_t>()
 	ucrtbase.dll!common_spawnlp<wchar_t>()
 	ucrtbase.dll!__wspawnlpe()
>	vcpkg.exe!vcpkg::System::cmd_execute_clean(const vcpkg::BasicCStringView<wchar_t> cmd_line={...}) Line 120
 	vcpkg.exe!vcpkg::Metrics::flush() Line 392
```
---

.exe, .pdb and .dmp in attach: [**dump.zip (Release x86)**](https://github.com/Microsoft/vcpkg/files/1226328/dump.zip)

**Comment by alexkaratarakis** [comment]
@Mixaill 
Please try the branch CreateProcess and see if it works for you. 
https://github.com/Microsoft/vcpkg/tree/CreateProcess

**Comment by Mixaill** [comment]
@alexkaratarakis Thanks, it works.

**Comment by alexkaratarakis** [comment]
Merged in master (26c8cb56afe89bf72e50b4490189a58de1d3f7cd).
